README.md

```diff
@@ -21,147 +24,131 @@ or to use in an npm script
+_please note: Currently, once a route, service, deployment config, build config, and imagestream config are created, those are re-used. The only thing that changes from deployment to deployment is the source code.  For application resources, you can update them by undeploying and then deploying again.  BuildConfigs and Imagestreams can be re-created using the --build.recreate flag_
 
-**deploy** - does everything, the archiving of code and everything that apply-resource does
+## Advanced Options
 
-**undeploy** - removes things that were deployed with the resource command
+While nodeshift tries to be very opinionated during the deployment process, but there are options available to pass to the cli or the API.
 
-**build** - archives the code, creates build config and imagestream and pushes the binary to the cluster
+#### version
+Outputs the current version of nodeshift
 
-#### Template Parameters
+#### projectLocation
+Changes the default location of where to look for your project. Defaults to your current working directory(CWD)
 
-Some templates might need to have a value set at "run time".  For example, in the template below, we have the `${SSO_AUTH_SERVER_URL}` parameter:
+#### configLocation
+Changes the default location of where to look for a configfile. Defaults to ~/.kube/
 
-        apiVersion: v1
-        kind: Deployment
-        metadata:
-            name: nodejs-rest-http-secured
-        spec:
-          template:
-            spec:
-              containers:
-                - env:
-                  - name: SSO_AUTH_SERVER_URL
-                    value: "${SSO_AUTH_SERVER_URL}"
-                  - name: REALM
-                    value: master
+#### osc.strictSSL
+Setting to pass to the [Openshift Rest Client](https://www.npmjs.com/package/openshift-rest-client) for SSL use.  To allow using a self-signed cert, set to false
 
-To set that using nodeshift, use the `-d` option with a KEY=VALUE, like this:
+#### osl.tryServiceAccount
+Setting to pass to the [Openshift Config Loader](https://www.npmjs.com/package/openshift-config-loader). Set to false to by-pass service account lookup or use the KUBERNETES_AUTH_TRYSERVICEACCOUNT environment variable
 
-    nodeshift -d SSO_AUTH_SERVER_URL=https://sercure-url
+#### nodeVersion
+Specify the version of Node.js to use for the deployed application. defaults to latest.  These version tags corespond to the docker hub tags of the [bucharest-gold s2i images](https://hub.docker.com/r/bucharestgold/centos7-s2i-nodejs/tags/)
 
-__note that we left off the "${}" on the key,  nodeshift knows to search for a key with ${} added back on__
+#### quiet
+supress INFO and TRACE lines from output logs.
 
-For more on writing openshift templates, [see here](https://docs.openshift.org/latest/dev_guide/templates.html#writing-templates)
+#### build.recreate
+Flag to recreate a buildConfig or Imagestream.  Defaults to false. Choices are "buildConfig", "imageStream", false, true
```

Comment:
If `true` are both the `BuildConfig` and `ImageStream` recreated? Also, capitalize `BuildConfig` in the first sentence.

Reply:
correct, a value of true will do both,  i'll make that more clear